src/Display/Collapsible/Collapsible.tsx

```diff
@@ -9,72 +10,63 @@ import {
   CollapsibleBody,
 } from '../../Style/Display/CollapsibleStyle';
 
-class Collapsible extends React.Component<Props, State> {
-  state = {
-    isOpen: true,
-  };
+const Collapsible: React.FunctionComponent<Props> = ({
+  label,
+  children,
+  className,
+  isOpen,
+  ...defaultProps
+}) => {
+  // If controlled, set initial state to isOpen prop
+  // If uncontrolled, set initial state to open
+  const [isOpenInternal, setIsOpenInternal] = React.useState<boolean>(
+    isUndefined(isOpen) ? true : isOpen
+  );
 
-  handleOpen = () => {
-    const { isOpen } = this.state;
-    this.setState({ isOpen: !isOpen });
-  };
-
-  componentDidMount() {
-    const { isOpen } = this.props;
-
-    if (isOpen === false && isOpen !== undefined) {
-      this.setState({
-        isOpen,
-      });
+  // If controlled, don't change the state on click
+  // If uncontrolled, toggle change
+  const toggle = React.useCallback(() => {
+    if (isUndefined(isOpen)) {
```

Comment:
I think it will be good for readability if we have a variable `const isUncontrolled = isUndefined(isOpen);`